ir/src/lib.rs

```diff
@@ -37,6 +37,10 @@ pub struct AirIR {
     //TODO: remove dead code attribute
     #[allow(dead_code)]
     constants: Constants,
+    // I don't think that adding a new field just for generic constraint evaluation worth it.
+    // TODO: get rid of this field
+    // amount of named columns: (main, aux)
+    num_polys: (u8, u8),
```

Comment:
I like that you're considering whether or not we want this, but I actually do think we need to add something here, because I don't think there's any other way to get this information. 

One note though - currently we only have 2 trace segments, but we've actually been making the IR more generic to support more than 2 segments, so this should probably be a vector to be consistent with those changes.